### PR TITLE
Docker integration

### PR DESCRIPTION
**Pull request by Chive**
Provides the functionality of setting up a local workspace for development through docker containers.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
     'requests>=1',
     'docopt',
     'pyyaml',
+    'docker-py',
```
Comment (stefanfoulis) [inline setup.py]: This should be an optional dependency.

**Comment by stefanfoulis** [comment]
Would be good if we show that ``--docker`` is an experimental option.

**Comment by Chive** [comment]
Adressed your points @stefanfoulis. When trying to use ``--docker`` w/o ``docker-py`` installed:
screen shot 2015-06-18 at 13 17 02

After installing it using ``pip install aldryn-client[docker]``
screen shot 2015-06-18 at 13 17 42

**Comment by Chive** [comment]
Adressed your points @stefanfoulis. When trying to use ``--docker`` w/o ``docker-py`` installed:
screen shot 2015-06-18 at 13 17 02

After installing it using ``pip install aldryn-client[docker]``
screen shot 2015-06-18 at 13 17 42

**Comment by Chive** [comment]
Ooops

**Pull request merged**